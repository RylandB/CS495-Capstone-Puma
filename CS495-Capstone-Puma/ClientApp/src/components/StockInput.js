import React from 'react'
import M from 'materialize-css'

let state = {
    assets: [],
    inputName: null,
    inputPrice: null,
    inputQuantity: null,
    inputExchangeType: null,
    inputEarningsPerShareDiluted: null,
    inputEarningPerShareBasic: null,
    inputEarningsPerShareEffectiveDate: null,
    inputPaymentFrequencyType: null,
    inputSharesOutstanding: null,
    inputIsIncludedIn13F: null,
    inputIsRestrictedByRule144A: null,
    inputCalculatedMarketCapType: null
};

export class StockInput extends React.Component{
    
    componentDidMount(){
        console.log("component did mount")
        M.AutoInit();
    }

    constructor(props) {
        super(props);
        this.state = state;
        
        this.addAsset = this.addAsset.bind(this);
        this.removeAsset = this.removeAsset.bind(this);
        this.handleInputChange = this.handleInputChange.bind(this);
    }

    componentWillUnmount() {
        state = this.state;
    }

    renderTableHeader() {
        if(this.state.assets.length>0) {
            let header = Object.keys(this.state.assets[0]);
            return header.map((key, index) => {
                return <th key={index}>{key.toUpperCase()}</th>
            })
        }    
    }

    renderTableData() {
        return this.state.assets.map((asset, index) => {
            const { name, price, quantity, exchangeType, earningsPerShareDiluted, earningsPerShareBasic, 
                earningsPerShareEffectiveDate, paymentFrequencyType, sharesOutstanding, isIncludedIn13F, 
                isRestrictedByRule144A, calculatedMarketCapType} = asset;
            return (
                <tr key={name}>
                    <td>{name}</td>
                    <td>{price}</td>
                    <td>{quantity}</td>
                    <td>{exchangeType}</td>
                    <td>{earningsPerShareDiluted}</td>
                    <td>{earningsPerShareBasic}</td>
                    <td>{earningsPerShareEffectiveDate}</td>
                    <td>{paymentFrequencyType}</td>
                    <td>{sharesOutstanding}</td>
                    <td>{isIncludedIn13F}</td>
                    <td>{isRestrictedByRule144A}</td>
                    <td>{calculatedMarketCapType}</td>
                </tr>
            )
        })
    }

    handleInputChange(event){

        const target = event.target;
        const value = event.target.value;
        const name = target.name;

        this.setState({
            [name]: value
        });

    }

    async addAsset(event){
        event.preventDefault();
        const newAsset = {
            name: this.state.inputName,
            price: this.state.inputPrice,
            quantity: this.state.inputQuantity,
            exchangeType: this.state.inputExchangeType,
            earningsPerShareDiluted: this.state.inputEarningsPerShareDiluted,
            earningsPerShareBasic: this.state.inputEarningPerShareBasic,
            earningsPerShareEffectiveDate: this.state.inputEarningsPerShareEffectiveDate,
            paymentFrequencyType: this.state.inputPaymentFrequencyType,
            sharesOutstanding: this.state.inputSharesOutstanding,
            isIncludedIn13F: this.state.inputIsIncludedIn13F,
            isRestrictedByRule144A: this.state.inputIsRestrictedByRule144A,
            calculatedMarketCapType: this.state.inputCalculatedMarketCapType
            
        };
        await this.setState({assets: this.state.assets.concat(newAsset)});
        this.props.stockCallback(this.state.assets);
    }

    async removeAsset(){
        await this.setState({asset: this.state.assets.pop()});
        this.props.stockCallback(this.state.assets);
    }

    render(){
        return(
            <div>
                <h3 id='title'>Stocks</h3>
                <table className='table' id='assets'>
                    <tbody>
                    <tr>{this.renderTableHeader()}</tr>
                    {this.renderTableData()}
                    </tbody>
                </table>
                <form onSubmit={this.addAsset}>
                    
                <div className="center-align">
                    <div className = "input-field col s6">    
                        <label>Name</label>
                        <input type="text" name="inputName" required onChange={this.handleInputChange} value={this.state.inputName}/>
                    </div>
                </div>
                
                <div className="center-align">
                    <div className = "input-field col s6">    
                        <label>Price</label>
                        <input type="number" name="inputPrice" required onChange={this.handleInputChange} value={this.state.inputPrice}/>
                    </div>
                </div>
                
                <div className="center-align">
                    <div className = "input-field col s6">    
                        <label>Quantity</label>
                        <input type="number" name="inputQuantity" required onChange={this.handleInputChange} value={this.state.inputQuantity}/>
                    </div>
                </div>
                
                <div className="center-align">
                    <div className = "input-field col s6">
                        <label>ExchangeType</label>
                        <input type="text" name="inputExchangeType" required onChange={this.handleInputChange} value={this.state.inputExchangeType}/>
                    </div>
                </div>
                
                <div className="center-align">
                    <div className = "input-field col s6">    
                        <label>EarningsPerShareDiluted</label>
                        <input type="number" name="inputEarningsPerShareDiluted" required onChange={this.handleInputChange} value={this.state.inputEarningsPerShareDiluted}/>
                    </div>
                </div>
                
                <div className="center-align">
                    <div className = "input-field col s6">    
                        <label>EarningPerShareBasic</label>
                        <input type="number" name="inputEarningPerShareBasic" required onChange={this.handleInputChange} value={this.state.inputEarningPerShareBasic}/>
                    </div>
                </div>
                
                <div className="center-align">
                    <div className = "input-field col s6">    
                        <label>EarningsPerShareEffectiveDate</label>
                        <input type="text" className="datepicker" name="inputEarningsPerShareEffectiveDate" required onChange={this.handleInputChange} value={this.state.inputEarningsPerShareEffectiveDate}/>
                    </div>
                </div>
                
                <div className="center-align">
                    <div className = "input-field col s6">    
                        <label>PaymentFrequencyType</label>
                        <input type="text" name="inputPaymentFrequencyType" required onChange={this.handleInputChange} value={this.state.inputPaymentFrequencyType}/>
                    </div>
                </div>
                
                <div className="center-align">
                    <div className = "input-field col s6">    
                        <label>SharesOutstanding</label>
                        <input type="number" name="inputSharesOutstanding" required onChange={this.handleInputChange} value={this.state.inputSharesOutstanding}/>
                    </div>
                </div>
                
                <div className="center-align">
                    <div className = "input-field col s6">    
                        <label>IsIncludedIn13F</label>
                        <select name="inputIsIncludedIn13F" required onChange={this.handleInputChange}  value={this.state.inputIsIncludedIn13F}>
                            <option value="true" >true</option>
                            <option value="false" >false</option>
                            <option value="null" selected="selected"> </option>
                        </select>
                    </div>
                </div>
                
                <div className="center-align">
                    <div className = "input-field col s6">    
                        <label>IsRestrictedByRule144A</label>
                        <select name="inputIsRestrictedByRule144A" required onChange={this.handleInputChange} value={this.state.inputIsRestrictedByRule144A}>
                            <option value="true" >true</option>
                            <option value="false" >false</option>
                            <option value="null" selected="selected"> </option>
                        </select>
                    </div>
                </div>
                
                <div className="center-align">
                    <div className = "input-field col s6">    
                        <label>CalculatedMarketCapType</label>
                        <input type="text" name="inputCalculatedMarketCapType" required onChange={this.handleInputChange} value={this.state.inputCalculatedMarketCapType}/>
                    </div>
                </div>
                    
                    <input type="submit" value="Add Asset" className="waves-effect waves-light btn-small"/>
                </form>
                <a onClick={this.removeAsset} className="waves-effect waves-light btn-small">Remove Asset</a>
                <br/>
                <br/>
            </div>
        );
    }

}