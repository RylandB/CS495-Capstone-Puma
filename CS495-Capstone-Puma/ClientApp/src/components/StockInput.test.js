import React from 'react';
import {StockInput} from "./StockInput";


it("PersonalInput Renders Correctly", () => {
    const wrapper = shallow(
        <StockInput />
    );

    expect(wrapper).toMatchSnapshot();

});