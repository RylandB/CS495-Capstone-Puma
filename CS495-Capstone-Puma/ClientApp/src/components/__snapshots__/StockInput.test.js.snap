// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalInput Renders Correctly 1`] = `
<div>
  <h3
    id="title"
  >
    Stocks
  </h3>
  <table
    className="table"
    id="assets"
  >
    <tbody>
      <tr />
    </tbody>
  </table>
  <form
    onSubmit={[Function]}
  >
    <label>
      Name
    </label>
    <input
      name="inputName"
      onChange={[Function]}
      required={true}
      type="text"
      value={null}
    />
    <br />
    <label>
      Price
    </label>
    <input
      name="inputPrice"
      onChange={[Function]}
      required={true}
      type="number"
      value={null}
    />
    <br />
    <label>
      Quantity
    </label>
    <input
      name="inputQuantity"
      onChange={[Function]}
      required={true}
      type="number"
      value={null}
    />
    <br />
    <label>
      ExchangeType
    </label>
    <input
      name="inputExchangeType"
      onChange={[Function]}
      required={true}
      type="text"
      value={null}
    />
    <br />
    <label>
      EarningsPerShareDiluted
    </label>
    <input
      name="inputEarningsPerShareDiluted"
      onChange={[Function]}
      required={true}
      type="number"
      value={null}
    />
    <br />
    <label>
      EarningPerShareBasic
    </label>
    <input
      name="inputEarningPerShareBasic"
      onChange={[Function]}
      required={true}
      type="number"
      value={null}
    />
    <br />
    <label>
      EarningsPerShareEffectiveDate
    </label>
    <input
      name="inputEarningsPerShareEffectiveDate"
      onChange={[Function]}
      required={true}
      type="date"
      value={null}
    />
    <br />
    <label>
      PaymentFrequencyType
    </label>
    <input
      name="inputPaymentFrequencyType"
      onChange={[Function]}
      required={true}
      type="text"
      value={null}
    />
    <br />
    <label>
      SharesOutstanding
    </label>
    <input
      name="inputSharesOutstanding"
      onChange={[Function]}
      required={true}
      type="number"
      value={null}
    />
    <br />
    <label>
      IsIncludedIn13F
    </label>
    <select
      name="inputIsIncludedIn13F"
      onChange={[Function]}
      required={true}
      value={null}
    >
      <option
        value="true"
      >
        true
      </option>
      <option
        value="false"
      >
        false
      </option>
      <option
        selected="selected"
        value="null"
      >
         
      </option>
    </select>
    <br />
    <label>
      IsRestrictedByRule144A
    </label>
    <select
      name="inputIsRestrictedByRule144A"
      onChange={[Function]}
      required={true}
      value={null}
    >
      <option
        value="true"
      >
        true
      </option>
      <option
        value="false"
      >
        false
      </option>
      <option
        selected="selected"
        value="null"
      >
         
      </option>
    </select>
    <br />
    <label>
      CalculatedMarketCapType
    </label>
    <input
      name="inputCalculatedMarketCapType"
      onChange={[Function]}
      required={true}
      type="text"
      value={null}
    />
    <br />
    <input
      type="submit"
      value="Add Asset"
    />
  </form>
  <button
    onClick={[Function]}
  >
     Remove Asset
  </button>
  <br />
  <br />
</div>
`;
